tests/test.rs

```diff
@@ -16,6 +18,7 @@ fn random_string(n: usize) -> String {
 }
 
 fn is_same(s: &String, fs: &FastString) -> bool {
+    // Why not just `s == fs` ?
```

Comment:
We want to check that is_empty and len also work

Reply:
Oh, riiight. For some reason I was thinking that short-circuting might play a trick on us here (ie, we return false due to is_empty check,  while `==` would've return `true`). But, clearly, we only short circuit on false, and and false leads to a test failure anyway. 

To put it another way, if theer's a bug in `len` (but not in is_empty or `==`), the current code uncovers it!